[Smart Lists] The Smart Lists tests should compile and run on iOS

rr-codes opened the pull request:
#### f3c2a539a44daa24e6a8c3cd48d10aa7f0052743
<pre>
[Smart Lists] The Smart Lists tests should compile and run on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=299182">https://bugs.webkit.org/show_bug.cgi?id=299182</a>
<a href="https://rdar.apple.com/160942503">rdar://160942503</a>

Reviewed by Abrar Rahman Protyasha and Ryosuke Niwa.

Support building and running the tests on iOS.

Also slightly refactor the Smart Lists code to better encapsulate related logic out of InsertTextCommand.cpp.

Canonical link: <a href="https://commits.webkit.org/300270@main">https://commits.webkit.org/300270@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db61500e1e9f1de73119e34526bdb1aad07a889f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121892 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41594 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32264 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128444 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73985 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4e5ea447-2bd7-4b21-9104-5b3e76d59f3b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123768 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42309 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50188 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92638 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61559 "Build is in progress. Recent messages:") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a8ab5d4e-cb08-4b20-994b-d6f93f10031f) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124844 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33739 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109157 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73297 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f97a88c3-9c34-47e5-bec8-38f60126357f) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32752 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27324 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71948 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103245 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27513 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131216 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48831 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37138 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101204 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49203 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105371 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101072 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25641 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46438 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24552 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45531 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48688 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54422 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48158 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51508 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49838 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/editing/TextListParser.cpp:
```diff
@@ -26,8 +26,22 @@
 #include "config.h"
 #include "TextListParser.h"
 
+#include <WebCore/CSSSerializationContext.h>
```
Why are we using `<WebCore/X.h>` header include in WebCore?
These should all be `"X.h"` instead.

rr-codes replied:
These should actually _never_ use quoted imports; once WebCore is fully modularized (which I should be able to enable very soon), the compiler will start enforcing that all private headers are included using `<WebCore/>`, since all private headers must be included this way to comply with modularization.

rniwa commented on Source/WebCore/editing/TextListParser.h:
```diff
@@ -26,12 +26,25 @@
+class AtomString;
+class String;
+class StringView;
+} // namespace WTF
```
Maybe just include `wtf/Forward.h`?

rr-codes committed:
[Smart Lists] The Smart Lists tests should compile and run on iOS
https://bugs.webkit.org/show_bug.cgi?id=299182
rdar://160942503

Reviewed by Abrar Rahman Protyasha and Ryosuke Niwa.

Support building and running the tests on iOS.

Also slightly refactor the Smart Lists code to better encapsulate related logic out of InsertTextCommand.cpp.

Canonical link: https://commits.webkit.org/300270@main
